lib/spack/spack/environment.py

```diff
@@ -445,23 +445,45 @@ def _write_yaml(data, str_or_file):
     syaml.dump_config(data, str_or_file, default_flow_style=False)
 
 
-def _eval_conditional(string):
-    """Evaluate conditional definitions using restricted variable scope."""
+def get_host_environment_metadata():
```

Comment:
it's a little weird for this to be in `environment.py` (which is about spack environments), and not in `spack.architecture` or something like that.  It looks like _get_host_environment was already here but maybe this stuff goes somewhere else.

Reply:
Yeah that's why I added it here, because it's alongside `_get_host_environment`. You want to move both?

Reply:
yeah I think both go either in `spack.util.environment` or in `spack.architecture`

Reply:
Moved to spack.util.environment.